pkg/scalers/splunk_o11y_scaler.go

```diff
@@ -0,0 +1,264 @@
+)
+
+type splunkO11yScaler struct {
+	metadata  *splunkO11yMetadata
+	apiClient *signalflow.Client
+	logger    logr.Logger
+}
+
+type splunkO11yMetadata struct {
+	query                string
+	queryValue           float64
+	queryAggegrator      string
+	activationQueryValue float64
+	metricName           string
+	vType                v2.MetricTargetType
+	accessToken          string
+	realm                string
+}
+
+func NewSplunkO11yScaler(ctx context.Context, config *scalersconfig.ScalerConfig) (Scaler, error) {
+	logger := InitializeLogger(config, "splunk_o11y_scaler")
+
+	meta, err := parseSplunkO11yMetadata(config, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Splunk metadata: %w", err)
+	}
+
+	apiClient, err := newSplunkO11yConnection(ctx, meta, config)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing Splunk Observability Cloud connection: %w", err)
+	}
+
+	return &splunkO11yScaler{
+		metadata:  meta,
+		apiClient: apiClient,
+		logger:    logger,
+	}, nil
+}
+
+func parseSplunkO11yMetadata(config *scalersconfig.ScalerConfig, logger logr.Logger) (*splunkO11yMetadata, error) {
+	meta := splunkO11yMetadata{}
+
+	// query
+	if query, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = query
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	// metric name
+	if metricName, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = GenerateMetricNameWithIndex(config.TriggerIndex, kedautil.NormalizeString(fmt.Sprintf("signalfx-%s", metricName)))
+	} else {
+		return nil, fmt.Errorf("no metric name given")
+	}
+
+	// queryValue
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %w", err)
+		}
+		meta.queryValue = queryValue
+	} else {
+		if config.AsMetricSource {
+			meta.queryValue = 0
+		} else {
+			return nil, fmt.Errorf("no queryValue given")
+		}
+	}
+
+	// activationQueryValue
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %w", err)
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	// queryAggregator
+	if val, ok := config.TriggerMetadata["queryAggregator"]; ok && val != "" {
+		queryAggregator := strings.ToLower(val)
+		switch queryAggregator {
+		case "max", "min", "avg":
+			meta.queryAggegrator = queryAggregator
+		default:
+			return nil, fmt.Errorf("queryAggregator value %s has to be one of 'max', 'min', or 'avg'.", queryAggregator)
+		}
+	} else {
+		meta.queryAggegrator = ""
+	}
+
+	// accessToken
+	if accessToken, ok := config.AuthParams["accessToken"]; ok {
+		// cleanedAccessToken := strings.ReplaceAll(accessToken, "\n", "")
+		// cleanedAccessToken = strings.ReplaceAll(accessToken, "\r", "")
```

Comment:
Is this cleaning still necessary?

Reply:
No, I should have done a better job with the clean up :(